types/if/bootproto.pp

```diff
@@ -0,0 +1,8 @@
+# Acceptable values for BOOTPROTO
+#
+# Only "dhcp" and "bootp" are used in /etc/sysconfig/network-scripts
+#
+# "static" and "none" are used in some parts of the module,
+# and those values have no side effects, so we accept them too.
```

Comment:
Not sure it's a good idea to keep allowing "static" and "none", as these values are not used by RedHat scripts.
But I kept them as I did not intend to introduce any regression in the PR.